File: RepoTransBench/target_projects/Java/Go/googlearchive_android-ClippingBasic/public_tests/messageonlylogfilter_public_test.go
package public_tests

import (
	"testing"

	"clippingbasic/tests"
)

type RecordingNode struct {
	LastPriority int
	LastTag      string
	LastMsg      string
	LastTr       error
}

func (n *RecordingNode) Println(priority int, tag, msg string, tr error) {
	n.LastPriority = priority
	n.LastTag = tag
	n.LastMsg = msg
	n.LastTr = tr
}

func setupFilter() (*tests.MessageOnlyLogFilter, *RecordingNode) {
	recorder := &RecordingNode{LastPriority: 1000, LastTag: "abc", LastMsg: "zzz"}
	filter := &tests.MessageOnlyLogFilter{}
	filter.SetNext(recorder)
	return filter, recorder
}

func TestPrintlnFiltersToMessageOnlyPublic(t *testing.T) {
	filter, recorder := setupFilter()
	filter.Println(tests.WARN, "PUBTAG", "public message", &customErr{"not seen"})
	if recorder.LastPriority != tests.NONE {
		t.Errorf("Expected priority NONE, got %d (Public)", recorder.LastPriority)
	}
	if recorder.LastTag != "" {
		t.Errorf("Expected empty tag, got '%s' (Public)", recorder.LastTag)
	}
	if recorder.LastMsg != "public message" {
		t.Errorf("Expected message 'public message', got '%s' (Public)", recorder.LastMsg)
	}
	if recorder.LastTr != nil {
		t.Errorf("Expected nil error (Public)")
	}
}

func TestConstructorWithNextPublic(t *testing.T) {
	recorder := &RecordingNode{}
	filter := tests.NewMessageOnlyLogFilter(recorder)
	if filter.GetNext() != recorder {
		t.Errorf("Expected Next to match recorder (Public)")
	}
}

func TestSetAndGetNextPublic(t *testing.T) {
	recorder := &RecordingNode{}
	f := &tests.MessageOnlyLogFilter{}
	f.SetNext(recorder)
	if f.GetNext() != recorder {
		t.Errorf("Expected Next to match recorder on set (Public)")
	}
}

func TestNullNextDoesNothingPublic(t *testing.T) {
	f := &tests.MessageOnlyLogFilter{}
	defer func() {
		if r := recover(); r != nil {
			t.Error("Should not panic when Next is nil (Public)")
		}
	}()
	f.Println(tests.VERBOSE, "v", "pqrs", nil)
}

type customErr struct{ msg string }
func (c *customErr) Error() string { return c.msg }

File: RepoTransBench/target_projects/Java/C#/googlearchive_android-ClippingBasic/public_tests/MessageOnlyLogFilterPublicTest.cs
using System;
using Xunit;
using ClippingBasicSample.Log;

namespace ClippingBasicSample.Tests.Public
{
    public class MessageOnlyLogFilterPublicTest
    {
        private class RecordingNode : ILogNode
        {
            public int LastPriority = int.MaxValue;
            public string LastTag = "abc";
            public string LastMsg = "zzz";
            public Exception LastTr;

            public void Println(int priority, string tag, string msg, Exception tr)
            {
                LastPriority = priority;
                LastTag = tag;
                LastMsg = msg;
                LastTr = tr;
            }
        }

        private MessageOnlyLogFilter _filter;
        private RecordingNode _recorder;

        public MessageOnlyLogFilterPublicTest()
        {
            _recorder = new RecordingNode();
            _filter = new MessageOnlyLogFilter();
            _filter.SetNext(_recorder);
        }

        [Fact]
        public void TestPrintlnFiltersToMessageOnlyPublic()
        {
            _filter.Println(Log.WARN, "PUBTAG", "public message", new Exception("not seen"));
            Assert.Equal(Log.NONE, _recorder.LastPriority);
            Assert.Null(_recorder.LastTag);
            Assert.Equal("public message", _recorder.LastMsg);
            Assert.Null(_recorder.LastTr);
        }

        [Fact]
        public void TestConstructorWithNextPublic()
        {
            var f2 = new MessageOnlyLogFilter(_recorder);
            Assert.Equal(_recorder, f2.GetNext());
        }

        [Fact]
        public void TestSetAndGetNextPublic()
        {
            var f3 = new MessageOnlyLogFilter();
            f3.SetNext(_recorder);
            Assert.Equal(_recorder, f3.GetNext());
        }

        [Fact]
        public void TestNullNextDoesNothingPublic()
        {
            var f4 = new MessageOnlyLogFilter();
            // Should not throw
            f4.Println(Log.VERBOSE, "v", "pqrs", null);
        }
    }
}